File: pyg_manager_web/src/main/webapp/js/service/specificationService.js
// 业务逻辑层 和后台数据交互 不出现$scope
app.service("specificationService",function ($http) {

    this.findSpecList=function () {
       return $http.get("../specification/findSpecList");
    }
    this.findAll=function () {
       return $http.get("../specification/findAll");
    }

    // 保存方法

    this.add=function (entity) {
       return $http.post("../specification/add",entity);
    }
    this.update=function (entity) {
        return $http.post("../specification/update",entity);
    }



    // 根据id查询对象
    this.findOne=function (id) {
       return $http.get("../specification/fineOne?id="+id);
    }


    // 删除
    this.dele=function (selectIds) {
           return $http.get("../specification/dele?ids="+selectIds);
    }


    this.search=function (pageNo,pageSize,searchEntity) {
        // $scope.searchEntity   页码  条数
        return $http.post("../specification/search?pageNo="+pageNo+"&pageSize="+pageSize,searchEntity);
    }

})

File: pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/SpecificationService.java
package com.pyg.sellergoods.service;

import bean.PageResult;
import com.pyg.pojo.TbSpecification;
import groupEntity.Specification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

    public List<TbSpecification> findAll();

    PageResult findPage(Integer pageNo, Integer pageSize);

    void add(Specification specification);

    Specification fineOne(Long id);

    void update(Specification specification);

    void dele(Long[] ids);

    PageResult findPage(Integer pageNo, Integer pageSize, TbSpecification tbSpecification);

    List<Map> findSpecList();
}
